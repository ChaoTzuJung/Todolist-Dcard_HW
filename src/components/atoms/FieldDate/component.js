import React, { Fragment, Component } from 'react';
import DatePicker from 'react-datepicker';

import styles from './index.css';
// CSS Modules, react-datepicker-cssmodules.css
import 'react-datepicker/dist/react-datepicker.css';

export class InputComponent extends Component {
	constructor(props) {
		super(props);
		this.handleChange = this.handleChange.bind(this);
	}

	handleChange(evt) {
		const { onChange = () => {} } = this.props;

		onChange(evt.target.value, evt.target.checked);
	}

	render() {
		const { name, value, content, className, small, onChange: _, ...other } = this.props;
		return (
			<label htmlFor={name} className={styles.label}>
				<input
					className={styles.input}
					type="text"
					id={name}
					name={name}
					value={value}
					onChange={this.handleChange}
					{...other}
				/>
			</label>
		);
	}
}

const FieldDate = ({ startTime, handleDateChange, dateOnly, timeOnly }) => (
	<Fragment>
		{!dateOnly && !timeOnly && (
			<DatePicker
				placeholderText="MMMM d, yyyy h:mm aa"
				selected={startTime}
				onChange={handleDateChange}
				minDate={new Date()}
				showTimeSelect
				timeFormat="HH:mm"
				timeIntervals={15}
				dateFormat="MMMM d, yyyy h:mm aa"
				timeCaption="time"
				customInput={<InputComponent />}
			/>
		)}
		{dateOnly && (
			<DatePicker
				placeholderText="yyyy/mm/dd"
				selected={startTime}
				onChange={handleDateChange}
				minDate={new Date()}
				dateFormat="yyyy/MM/dd"
				customInput={<InputComponent />}
			/>
		)}
		{timeOnly && (
			<DatePicker
				placeholderText="HH:mm"
				selected={startTime}
				onChange={handleDateChange}
				minDate={new Date()}
				showTimeSelect
				showTimeSelectOnly
				timeIntervals={60}
				dateFormat="HH:mm"
				timeFormat="HH:mm"
				timeCaption="Time"
				customInput={<InputComponent />}
			/>
		)}
	</Fragment>
);

export default FieldDate;
