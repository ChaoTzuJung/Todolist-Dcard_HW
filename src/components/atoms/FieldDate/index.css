.label {
	& > input {
		padding: 8px 16px;
		width: 162px;
		background-color: #fff;
		border-radius: 2px;
		color: #000;
		font-weight: 300;
		font-size: 16px;
		border: none;

		&::placeholder {
			color: #cbcbcb;
			font-size: 16px;
			font-weight: 300;
		}
	}
}
