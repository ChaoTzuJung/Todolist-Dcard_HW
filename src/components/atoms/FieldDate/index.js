import { contain } from 'react-container-helper';

import FieldDate from './component';

const initState = () => ({
	startDate: Date.now(),
});

const mapSetStateToProps = ({ startDate }, { className, startTime, onChange, handleDateChange, ...other }, setState) => ({
	// state
	startDate,

	// props
	className,
	startTime,
	onChange,
	handleDateChange,
	...other,

	// actions
	handleChange(e) {
		setState({ startDate: e });
		onChange(e);
	},
});

export default contain(initState, mapSetStateToProps)(FieldDate);
