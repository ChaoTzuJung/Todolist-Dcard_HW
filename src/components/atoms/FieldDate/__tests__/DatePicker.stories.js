import React from 'react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import FieldDate from 'components/atoms/FieldDate';

const stories = storiesOf('Commons|atoms/FieldDate', module);

stories.addDecorator(withKnobs);

stories.add('with default', () => (
	<div style={{ margin: '16px' }}>
		<FieldDate onChange={action('selected date and time')} startTime={null} />
	</div>
));

stories.add('with date only', () => (
	<div style={{ margin: '16px' }}>
		<FieldDate dateOnly onChange={action('selected date')} startTime={null} />
	</div>
));

stories.add('with time only', () => (
	<div style={{ margin: '16px' }}>
		<FieldDate timeOnly onChange={action('selected time')} startTime={null} />
	</div>
));
