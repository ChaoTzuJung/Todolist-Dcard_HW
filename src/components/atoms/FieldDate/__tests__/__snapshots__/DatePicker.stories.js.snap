// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with date only 1`] = `
<div
  style={
    Object {
      "margin": "16px",
    }
  }
>
  <Contain(FieldDate)
    dateOnly={true}
    onChange={[Function]}
    startTime={null}
  >
    <FieldDate
      dateOnly={true}
      handleChange={[Function]}
      onChange={[Function]}
      startDate={null}
      startTime={null}
    >
      <DatePicker
        allowSameDay={false}
        customInput={<InputComponent />}
        dateFormat="yyyy/MM/dd"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        minDate={2019-04-13T05:46:56.213Z}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        placeholderText="yyyy/mm/dd"
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={null}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={false}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
      >
        <PopperComponent
          hidePopper={true}
          popperComponent={null}
          popperModifiers={
            Object {
              "preventOverflow": Object {
                "boundariesElement": "viewport",
                "enabled": true,
                "escapeWithReference": true,
              },
            }
          }
          popperPlacement="bottom-start"
          popperProps={Object {}}
          targetComponent={
            <div
              className="react-datepicker__input-container"
            >
              <InputComponent
                className=""
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="yyyy/mm/dd"
                readOnly={false}
                value=""
              />
            </div>
          }
        >
          <Manager>
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <InputComponent
                      className=""
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="yyyy/mm/dd"
                      readOnly={false}
                      value=""
                    >
                      <label
                        className="label"
                      >
                        <input
                          className="input"
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="yyyy/mm/dd"
                          readOnly={false}
                          type="text"
                          value=""
                        />
                      </label>
                    </InputComponent>
                  </div>
                </div>
              </InnerReference>
            </Reference>
          </Manager>
        </PopperComponent>
      </DatePicker>
    </FieldDate>
  </Contain(FieldDate)>
</div>
`;

exports[`with default 1`] = `
<div
  style={
    Object {
      "margin": "16px",
    }
  }
>
  <Contain(FieldDate)
    onChange={[Function]}
    startTime={null}
  >
    <FieldDate
      handleChange={[Function]}
      onChange={[Function]}
      startDate={null}
      startTime={null}
    >
      <DatePicker
        allowSameDay={false}
        customInput={<InputComponent />}
        dateFormat="MMMM d, yyyy h:mm aa"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        minDate={2019-04-13T05:46:56.172Z}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        placeholderText="MMMM d, yyyy h:mm aa"
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={null}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={true}
        strictParsing={false}
        timeCaption="time"
        timeFormat="HH:mm"
        timeInputLabel="Time"
        timeIntervals={15}
        withPortal={false}
      >
        <PopperComponent
          hidePopper={true}
          popperComponent={null}
          popperModifiers={
            Object {
              "preventOverflow": Object {
                "boundariesElement": "viewport",
                "enabled": true,
                "escapeWithReference": true,
              },
            }
          }
          popperPlacement="bottom-start"
          popperProps={Object {}}
          targetComponent={
            <div
              className="react-datepicker__input-container"
            >
              <InputComponent
                className=""
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="MMMM d, yyyy h:mm aa"
                readOnly={false}
                value=""
              />
            </div>
          }
        >
          <Manager>
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <InputComponent
                      className=""
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="MMMM d, yyyy h:mm aa"
                      readOnly={false}
                      value=""
                    >
                      <label
                        className="label"
                      >
                        <input
                          className="input"
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="MMMM d, yyyy h:mm aa"
                          readOnly={false}
                          type="text"
                          value=""
                        />
                      </label>
                    </InputComponent>
                  </div>
                </div>
              </InnerReference>
            </Reference>
          </Manager>
        </PopperComponent>
      </DatePicker>
    </FieldDate>
  </Contain(FieldDate)>
</div>
`;

exports[`with time only 1`] = `
<div
  style={
    Object {
      "margin": "16px",
    }
  }
>
  <Contain(FieldDate)
    onChange={[Function]}
    startTime={null}
    timeOnly={true}
  >
    <FieldDate
      handleChange={[Function]}
      onChange={[Function]}
      startDate={null}
      startTime={null}
      timeOnly={true}
    >
      <DatePicker
        allowSameDay={false}
        customInput={<InputComponent />}
        dateFormat="HH:mm"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        minDate={2019-04-13T05:46:56.224Z}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        placeholderText="HH:mm"
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={null}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={true}
        showTimeSelectOnly={true}
        strictParsing={false}
        timeCaption="Time"
        timeFormat="HH:mm"
        timeInputLabel="Time"
        timeIntervals={60}
        withPortal={false}
      >
        <PopperComponent
          hidePopper={true}
          popperComponent={null}
          popperModifiers={
            Object {
              "preventOverflow": Object {
                "boundariesElement": "viewport",
                "enabled": true,
                "escapeWithReference": true,
              },
            }
          }
          popperPlacement="bottom-start"
          popperProps={Object {}}
          targetComponent={
            <div
              className="react-datepicker__input-container"
            >
              <InputComponent
                className=""
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="HH:mm"
                readOnly={false}
                value=""
              />
            </div>
          }
        >
          <Manager>
            <Reference>
              <InnerReference
                setReferenceNode={[Function]}
              >
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <InputComponent
                      className=""
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="HH:mm"
                      readOnly={false}
                      value=""
                    >
                      <label
                        className="label"
                      >
                        <input
                          className="input"
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="HH:mm"
                          readOnly={false}
                          type="text"
                          value=""
                        />
                      </label>
                    </InputComponent>
                  </div>
                </div>
              </InnerReference>
            </Reference>
          </Manager>
        </PopperComponent>
      </DatePicker>
    </FieldDate>
  </Contain(FieldDate)>
</div>
`;
